src/java/org/apache/cassandra/db/memtable/pmem/MemoryBlockDataOutputPlus.java

```diff
@@ -162,9 +169,12 @@ public void writeUTF(String s) throws IOException
     {
         try
         {
-            UnbufferedDataOutputStreamPlus.writeUTF(s, this);
+            int strlen = s.length();
+            writeShort(strlen);
```

Comment:
For this we had to duplicate some of the writeUTF function because we were running into a little-endian issue otherwise while reading back the size of the string